lib/roo/excelx/sheet.rb

```diff
@@ -113,6 +109,35 @@ def pad_cells(cell, last_column)
         (cell.coordinate.column - 1 - last_column).times { pad << nil }
         pad
       end
+
+      def first_last_row_col
```

Comment:
Something about this implementation feels like its doing too much work. For first row/column, can't we just find the first non-empty row/non-empty column then  break?

For last row/column can't we do the same thing in reverse? Iterating over every cell seems incredible wasteful. Maybe something like

```ruby
def first_row_col
  first_row = first_col = nil
  cells.each do |(row, col) cell|
    next if cell && !cell.empty?
    first_row = row
    first_col = col
    break
  end

  @first_row = first_row
  @first_col = first_col
end

def last_row_col
  last_row = last_col = nil
  cells.reverse_each do |(row, col) cell|
    next if cell && !cell.empty?
    last_row = row
    last_col = col
    break
  end

  @last_row = last_row
  @last_col = last_col
end
```
Obviously that isn't exactly right, but it seems like a much simpler (and faster) way of doing this.

Side note: `reverse_each` is different that `Array.reverse.each` in case there was any confusion there.